Discussion (#10)
greenelab/generalization-manuscript: content/02.main-text.md

**D.** Distribution of performance comparisons between "best" and "smallest good" model selection strategies, for TCGA -> CCLE generalization with neural network hidden layer size as the regularization axis. Positive x-axis values indicate better performance for the "best" model, negative values indicate better performance for the "smallest good" model.
](images/figure_5.png){#fig:tcga_ccle_nn width="90%"}

## Discussion

Using public cancer genomics and transcriptomics data from TCGA and CCLE, we studied generalization of mutation status classifiers for a wide variety of cancer driver genes.
We designed experiments to evaluate generalization across biological contexts by holding out cancer types in TCGA, and to evaluate generalization across datasets by training models on TCGA and evaluating them on CCLE, and vice-versa.

> Our results demonstrated that in general, smaller or more parsimonious models do not tend to generalize more effectively across cancer types or across datasets, and in the absence of prior knowledge about a prediction problem, simply choosing the model that performs the best on a holdout dataset is at least as effective for selecting models that generalize.

Reviewer: ```suggestion
We found that, in general, smaller or more parsimonious models do not tend to generalize more effectively across cancer types or across datasets, and in the absence of prior knowledge about a prediction problem, simply choosing the model that performs the best on a holdout dataset is at least as effective for selecting models that generalize.
```

> Our data suggest that this conclusion holds in both linear models (LASSO logistic regression) and non-linear deep neural networks with hidden layer size as the regularization parameter of interest.

Reviewer: ```suggestion
This was apparent in both linear models (LASSO logistic regression) and non-linear deep neural networks when using hidden layer size as the regularization parameter of interest.
```

Reviewer: I might move your final paragraph up here and make this the first part of that paragraph (or remove if redundant).

In our non-linear model experiments, we did not observe better generalization across datasets for fully connected neural networks with fewer hidden layer nodes, and our preliminary results indicated a similar trend for dropout and weight decay.
Compared to linear models, it is less clear how to define a "small" or "parsimonious" neural network model since there are many regularization techniques that one may use to control complexity.
Rather than simply removing nodes and keeping the network fully connected, another approach to parsimony could be to select an inductive bias to guide the size reduction of the network.
Existing examples include network structures guided by protein-protein interaction networks or function/pathway ontologies [@doi:10.1016/j.ccell.2020.09.014; @doi:10.1093/bioinformatics/btx624; @doi:10.1186/s13059-020-02100-5; @doi:10.1016/j.patter.2022.100565].
It is possible that a smaller neural network with a structure that corresponds more appropriately to the prediction problem would achieve better generalization results, although choosing an apt network structure or data source can be a challenging aspect of such efforts.

> 

Reviewer: Final paragraph (or conclusions section if the format has one):

* It is tempting to assume that the simplest model will generalize best, but we don't observe that.
* There can still be reasons to train small models and/or the best model of a certain size (e.g., clinical biomarker practicality, etc).
* If the goal is generalization, we recommend directly evaluating that. If it's not available, we recommend choosing the model that performs the best on unseen data.

Author: I'll add this as a conclusions section/separate PR for now since we haven't decided where we're submitting, but we can always merge into the discussion if the journal wants it.
